security_scheme.go

```diff
@@ -0,0 +1,89 @@
+package spec3
+
+import (
+	"encoding/json"
+
+	"github.com/go-openapi/spec"
+	"github.com/go-openapi/swag"
+)
+
+// SecurityScheme defines reusable Security Scheme Object
+// as described in https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#securitySchemeObject
+type SecurityScheme struct {
+	// Refable is a simple object to allow referencing other components in the specification, more at https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#referenceObject
+	spec.Refable
+	// SecuritySchemeProps defines a security scheme that can be used by the operations
+	SecuritySchemeProps
+	// VendorExtensible allows for adding additional fields, more at https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#specification-extensions
+	spec.VendorExtensible
+}
+
+// MarshalJSON is a custom marshal function that knows how to
+// encode SecurityScheme as JSON
+func (s *SecurityScheme) MarshalJSON() ([]byte, error) {
+	b1, err := json.Marshal(s.SecuritySchemeProps)
+	if err != nil {
+		return nil, err
+	}
+	b2, err := json.Marshal(s.VendorExtensible)
+	if err != nil {
+		return nil, err
+	}
+	b3, err := json.Marshal(s.Refable)
+	if err != nil {
+		return nil, err
+	}
+	return swag.ConcatJSON(b1, b2, b3), nil
```

Comment:
i see it's the same way of [de]serializing model as v2 does. just out of curiosity, won't `json:",inline"` does achieve the same effect?

Reply:
i'm not sure, let me check that . . .

Reply:
No, it doesn't. This hack is to decode/encode the `x-*` vendor extensions into a map. `,inline` is for inline structs. Moreover, `,inline` has no effect whatsoever. It's just cargo culted from us using some old yaml library in 2014.